master/internal/core.go

```diff
@@ -613,6 +613,19 @@ func (m *Master) rwCoordinatorWebSocket(socket *websocket.Conn, c echo.Context)
 	return actorRef.AwaitTermination()
 }
 
+func updateClusterHeartbeat(db *db.PgDB) {
+	t := time.NewTicker(10 * time.Minute)
+	defer t.Stop()
+	current_cluster_heartbeat := time.Now().UTC().Truncate(time.Millisecond)
+	for range t.C {
+		err := db.UpdateClusterHeartBeat(current_cluster_heartbeat)
+		if err != nil {
+			log.Error(err.Error())
+		}
+
+	}
```

Comment:
to get the first tick immediately
```suggestion
	t := time.NewTicker(10 * time.Minute)
	defer t.Stop()
	current_cluster_heartbeat := time.Now().UTC().Truncate(time.Millisecond)
	for {
		err := db.UpdateClusterHeartBeat(current_cluster_heartbeat)
		if err != nil {
			log.Error(err.Error())
		}
		<-t.C
	}
```

Reply:
also just because making sure we can gracefully shutdown is generally useful, can you go ahead pass a `context.Context` to this?

Reply:
done!